pennylane/ops/op_math/sum.py

```diff
@@ -220,6 +221,18 @@ def matrix(self, wire_order=None):
 
         return math.expand_matrix(reduced_mat, sum_wires, wire_order=wire_order)
 
+    def compute_grouping(self, grouping_type="qwc", method="rlf"):
+        """docs"""
+        if _hash := self.hash not in Sum._grouping_cache:
+            with qml.QueuingManager.stop_recording():
+                obs_groups = qml.pauli.group_observables(
+                    self.operands, grouping_type=grouping_type, method=method
+                )
+
+            Sum._grouping_cache[_hash] = obs_groups
+
+        return Sum._grouping_cache[_hash]
```

Comment:
This is so that the cache is "global". Whenever someone calls `compute_grouping` for a `Sum` with the same hash as another `Sum` for which we already computed groupings, rather than doing the grouping again, we can just return the cached groupings.

Reply:
Removed from this PR. Caching functionality for grouping will be added in another PR.

Reply:
Just out of curiosity still, how does that work when you have two Sum instances with different groupings? Because they have different hashes its okay and they dont collide?

Reply:
Yep. that's the plan. `qml.Hermitian` uses a similar way to cache its eigendecomposition. Instead of the hash though, it uses its flattened matrix as the key.